sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -19,33 +25,26 @@ public final class EmbeddingItem {
      * List of embeddings value for the input prompt. These represent a measurement of the
      * vector-based relatedness of the provided input.
      */
-    @Generated
     @JsonProperty(value = "embedding")
-    private List<Double> embedding;
+    private BinaryData embedding;
 
     /**
-     * Creates an instance of EmbeddingItem class.
+     * Get the embedding property: List of embeddings value for the input prompt. These represent a measurement of the
+     * vector-based relatedness of the provided input.
      *
-     * @param embedding the embedding value to set.
-     * @param promptIndex the promptIndex value to set.
+     * @return the embedding value.
      */
-    @Generated
-    @JsonCreator
-    private EmbeddingItem(@JsonProperty(value = "embedding") List<Double> embedding,
-        @JsonProperty(value = "index") int promptIndex) {
-        this.embedding = embedding;
-        this.promptIndex = promptIndex;
+    public List<Float> getEmbedding() {
+        return convertToFloatList(embedding.toString());
```

Comment:
We should not convert each time this method is called. This should be done only once.